src/stp/new_tactics/DriveWithBall.cpp

```diff
@@ -10,58 +10,44 @@
 namespace rtt::ai::stp::tactic {
 
 DriveWithBall::DriveWithBall() {
-    // Create state machine of skills and initialize first skill
-    // TODO: The rotate skill might be unnecessary in this tactic if our dribbler works well whilst driving backwards
-    skills = rtt::collections::state_machine<Skill, Status, StpInfo>{skill::Rotate(), skill::GoToPos()};
-}
-
-void DriveWithBall::onInitialize() noexcept {}
-
-void DriveWithBall::onUpdate(Status const& status) noexcept {}
-
-void DriveWithBall::onTerminate() noexcept {
-    // Call terminate on all skills
-    for (auto& x : skills) {
-        x->terminate();
-    }
+  // Create state machine of skills and initialize first skill
+  // TODO: The rotate skill might be unnecessary in this tactic if our dribbler works well whilst driving backwards
+  skills = rtt::collections::state_machine<Skill, Status, StpInfo>{skill::Rotate(), skill::GoToPos()};
 }
 
 std::optional<StpInfo> DriveWithBall::calculateInfoForSkill(StpInfo const& info) noexcept {
-    StpInfo skillStpInfo = info;
+  StpInfo skillStpInfo = info;
 
-    if(!skillStpInfo.getPositionToMoveTo() || !skillStpInfo.getBall()) return std::nullopt;
+  if (!skillStpInfo.getPositionToMoveTo() || !skillStpInfo.getBall()) return std::nullopt;
 
-    double angleToBall = (info.getPositionToMoveTo().value() - info.getBall()->get()->getPos()).angle();
-    skillStpInfo.setAngle(angleToBall);
+  double angleToBall = (info.getPositionToMoveTo().value() - info.getBall()->get()->getPos()).angle();
+  skillStpInfo.setAngle(angleToBall);
 
-    // When driving with ball, we need to activate the dribbler
-    // For now, this means full power, but this might change later
-    // TODO: TUNE better way to determine dribblerspeed
-    skillStpInfo.setDribblerSpeed(100);
+  // When driving with ball, we need to activate the dribbler
+  // For now, this means full power, but this might change later
+  // TODO: TUNE better way to determine dribblerspeed
+  skillStpInfo.setDribblerSpeed(100);
 
-    return skillStpInfo;
+  return skillStpInfo;
 }
 
 bool DriveWithBall::isTacticFailing(const StpInfo& info) noexcept {
-    // Fail if we don't have the ball or there is no movement position
-    return !info.getRobot()->hasBall() || !info.getPositionToMoveTo();
+  // Fail if we don't have the ball or there is no movement position
+  return !info.getRobot()->hasBall() || !info.getPositionToMoveTo();
 }
 
 bool DriveWithBall::shouldTacticReset(const StpInfo& info) noexcept {
-    // Should reset if the angle the robot is at is no longer correct
-    auto robotAngle = info.getRobot()->get()->getAngle();
-    auto ballToRobotAngle = (info.getBall()->get()->getPos() - info.getRobot()->get()->getPos()).angle();
-
-    return fabs(robotAngle + Angle(ballToRobotAngle)) <= stp::control_constants::GO_TO_POS_ANGLE_ERROR_MARGIN;
```

Comment:
I don't think this preserves the meaning

Reply:
I will revert this then!